projects/plugins/boost/app/lib/minify/functions-service.php

```diff
@@ -336,6 +282,39 @@ function ( $match ) use ( $dirpath, &$pre_output ) {
 	);
 }
 
+function jetpack_boost_page_optimize_get_file_paths( $args ) {
+	$paths = File_Paths::get( $args );
+	if ( $paths ) {
+		$args = $paths->get_paths();
+	} else {
+		// Kept for backward compatibility in case cached page is still referring to old formal asset URLs. Remove after $$next-version$$.
+
+		// It's a base64 encoded list of file path.
+		// e.g.: /_jb_static/??-eJzTT8vP109KLNJLLi7W0QdyDEE8IK4CiVjn2hpZGluYmKcDABRMDPM=
+		if ( '-' === $args[0] ) {
+
+			// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged,WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_decode
+			$args = @gzuncompress( base64_decode( substr( $args, 1 ) ) );
+		}
+
+		// It's an unencoded comma separated list of file paths.
+		// /foo/bar.css,/foo1/bar/baz.css?m=293847g
+		$version_string_pos = strpos( $args, '?' );
+		if ( false !== $version_string_pos ) {
+			$args = substr( $args, 0, $version_string_pos );
+		}
+		// /foo/bar.css,/foo1/bar/baz.css
+		$args = explode( ',', $args );
```

Comment:
I am unsure if this is needed. The files are also cached for 24 hours ones they are generated. This block of code allows the minifier to generate the files again in case they are missing. Let me know what you think.

Reply:
I think we should have this.

I've removed the `deprecated` part, since I don't think we should ever remove it. I want to avoid situations where a cached site gets Boost updated in the background and it tries to run functionality that doesn't exist.